src/components/Layout/Topbar/TopBar.jsx

```diff
@@ -0,0 +1,74 @@
+import { HStack, Spacer, Icon, Heading, Button, Flex, Text } from '@chakra-ui/react';
+import PropTypes from 'prop-types'; // ES6
+import { FaTeamspeak } from 'react-icons/fa';
+import { Link } from 'react-router-dom';
+
+function TopBar({ isLogged, setIsLogged }) {
+  let links;
+
+  const handleLogout = () => {
+    setIsLogged(false);
+  };
+
+  const makeLinkButton = (el, index) => {
+    return (
+      <Link
+        key={index}
+        to={`/${el}`}
+        onClick={() => {
+          if (el === 'logout') handleLogout();
+        }}
+      >
+        <Button
+          m=".2rem 0 .2rem 1rem"
+          bg="color.buttonBlue"
+          color="color.whiteText"
+          _hover={{
+            background: 'color.buttonBlueHover',
+          }}
+          mr="1rem"
+        >
+          <Text fontSize="2xl">{el}</Text>
+        </Button>
+      </Link>
+    );
+  };
+
+  if (!isLogged) {
+    links = ['login', 'register'].map((el, i) => makeLinkButton(el, i));
```

Comment:
We don't have to render the items to jsx here, we can do it in the return statement below, so all jsx is together

Here we can just do

```js
const guestLinks = ['login', 'register'];
const userLinks = ['logout', 'my-profile']
const links = isLogged ? userLinks : guestLinks;

return (
    {links.map(link => <TopBarLink {...link} />}
)
```